### ISSUE TITLE
Quick Download

### ISSUE THREAD

**Issue by jiwachow** [modrinth/knossos#926]
### Is your feature request related to a problem? Please describe.

I am trying to build a new instance for myself while selecting the filters version, and loader. 
I am trying to download all mods that are compatible with my filters, and their dependencies and optional mods. 
I am trying to do it as fast as possible. 
I am trying to do it without opening each mod page, and without opening each dependency and optional mod page.

Currently, the process of downloading multiple mods from the Modrinth website is time-consuming and requires opening multiple pages and manually selecting options. This leads to a slow and tedious experience for the user.

**Problem:**
I'm trying to build up new instance for myself while selecting the filters version, and loader.
I'm trying to download all mods that are compatible with my filters, and their dependencies and optional mods.
I'm trying to do it as fast as possible.
I'm trying to do it without opening each mod page, and without opening each dependency and optional mod page.

**Process now:** 
-Optional: select filters for the list (ex: 1.19.2, Quilt),
-browse the list,
-open each page,
-open see all versions page,
-filter quilt, 
-filter 1.19.2, 
-search for visually and open the latest correct version,
-find the download button and click it,
-open dependencies and download,
-open optional and download
(repeat for all mods I want to download, and for all dependencies and optional)

**Process as expected:**
-Optional: select filters for the list (ex: 1.19.2, Quilt), or use default filters (account-configurable) (ex: latest version, Quilt),
-Optional: change download button behaviour(see below), or use default behaviour (ex: download mod only)
-browse the list,
-click download button directly on the list page, without opening the mod page
(repeat for all mods I want to download)

**Effect**
It would allow fastest usage of the website, and it would allow to use the website without using unnecessary traffic.
Users would be able to download all mods they want, without having to open each mod page, and without having to open each dependency and optional mod page.

### Describe the solution you'd like

I would like a "Quick Download" button to be added to the mod list page, which allows users to quickly download all mods that are compatible with their selected filters and their dependencies and optional mods. This button should be available when a user is logged in and their account is using the mod list.

**Suggestion:**
Make it so, that the when **logged in account** is using Mods list, it provides quick download button (instead of download icon) according to [<default_filters>](https://github.com/modrinth/knossos/issues/924), or filter selection.

**Quick download button options:**
There should be additional option "what to download", on top of the modlist, either two Booleans, or a list.

Two bools:
-download dependencies
-download optional

List:
download mod
download mod with dependencies
download mod with optional
download all

**Followed only filter:**
There could be additional filter, [<followed_only>](https://github.com/modrinth/knossos/issues/923) in filters panel, that would allow filter out mods that person doesn't follow.

**Result:**
Downloads the: 
(<is_followed>) <selected_version> <selected_loader> (<should_dependencies_be_downloaded>) (<should_optional_be_downloaded>) 
immediately, when the download button is clicked without having to enter the mod page. 
<selected_version> and <selected_loader> are mandatory, 
<should_dependencies_be_downloaded>, <should_optional_be_downloaded> and <is_followed> are optional.

For multiple downloads (dependencies and optional), it would be best to start multiple file download instead of zipping them since it's easier to use later on. 

**Version and Loader filter:**
There needs to be a version and loader chosen by default, or configured when account is created, if you want to make browsing the website ultimately fast.
Using [<default_filters>](https://github.com/modrinth/knossos/issues/925), would help a lot.

This won't work if bug [<bug_report>](https://github.com/modrinth/knossos/issues/922) is not fixed.

**Alternatively** without defaults, the quick download button should be unclickable by default, and display message on cursor hover, that says: "Please select a loader to quick download", and when loader is selected, it should be clickable. Version would be latest.

Why I don't like this alternative? - Because it would be frustrating multiple users, when they inevitably download latest mod version, and it doesn't work, since latest mod version is not always compatible with newest minecraft patch.

Why not use the most popular loader when no loader filter is selected? - Because for one mod it might be Fabric, for another it might be Forge, and for another it might be Quilt, and it would be frustrating for the user, and not the most efficient use of time and traffic.
 
Why not make it so that it downloads all loaders? - Because it would be a waste of time and traffic, and it would be a waste of time for the user, since he would have to delete the files he doesn't need.

### Describe alternatives you've considered

An alternative solution would be to have a [<download_all_filtered>](https://github.com/modrinth/knossos/issues/925) button, which allows users to download all filtered mods in a zip file, but this feature is different and has a different use case.

### Additional context

- The "Quick Download" button should have settings next to sorting for "what to download" such as:
-download mod only, 
-download mod and dependencies, 
-download mod and optional, and 
-download all.
This can be done either with two Booleans, or with list.

- It would be nice to have [<followed_only>](https://github.com/modrinth/knossos/issues/923) filter as an additional option here.

- The "Quick Download" button should take into account the user's selected filters, or their default filters (account-configurable) as specified in [<default_filters>](https://github.com/modrinth/knossos/issues/924).

- The "Quick Download" button should be provided with double confirmation, and it should allow to change the options, without having to go back to the list.

- The "Quick Download" button should be provided with a link to the [<default_filters>](https://github.com/modrinth/knossos/issues/924)  settings menu, which is accessible from the account settings, or from "defaults" button in the filters pane.

- Bug [<bug_report>](https://github.com/modrinth/knossos/issues/922) should be fixed first for this feature to be possible.

**Comment by jiwachow** [comment]
> Duplicate of #157 and of [modrinth/.github#30](https://github.com/orgs/modrinth/discussions/30) (shopping cart functionality)

Disagree, this is not download all, or download collection.

This is quick download single mod, from the list, as discussed today on discord, optionally with dependencies, optionally with optionals.

**Comment by triphora** [comment]
The main issue is that we don't want people to be able to download stuff without first reading the description and project page. While a very small number of power-users would appreciate it, it would be abused by most users. What you could do with collections is have a collection for your use only that assists you in updating your modpack, since the functionality would allow you to bulk-download the mods in the collection for a specified loader/version combination.

**Comment by jiwachow** [comment]
Yeah, but then you'd have to manage collections which is cumbersome, especially when doing something odd, or if there are new mods coming in. 
This feature is the one I dream of, and I'm sure many modders would love it too. 
Why do you want to force people to read? Your answer frustrates me.
There is lot of reasons why I wouldn't want to read descriptions, like for instance I know them already, from testing 5 different instances on different loaders in last 3 days. 
It's the power users, people who make instances for their kids, and are working regular jobs, and modpack creators that are spending most time on this, and it would save them tremendous amount of time. 

If you want non power-users to have this turned off, just add it as advanced option in user account settings, and it would be no issue for your dream of making people read descriptions. But just closing the issue, without consideration hurts me.

**Comment by triphora** [comment]
> Why do you want to force people to read?

Bad things happen when people blindly download things. Modrinth should not be considered a 100% trusted place to blindly download stuff from, even if we want it to be that way, no site is able to say that it is 100% trusted.

I do like your idea of making it disabled by default, so I will reopen the issue, but do note that it is very low priority

**Comment by jiwachow** [comment]
I put real thought into it, and discussed this as useful issue on your discord, then created a 4 issue 1 bug solution, that is universal, simple, and cheap, without you having to add a whole new interface, like catalogue. And without the need to set-up.

In total, let's count:
-one setting next to sorting (download dependencies/optional)
-one settings link (defaults) and,
-one filter (followed only) in filter pane
-one backend change to save current session filters as default filters, 
-one settings panel in account settings (default filter)
-one backend table, from 6 proposed solutions, to solve the bug
-one backend table, (or proposition to use the same from the bug solution), for follow only filter
-one icon changed into a download link from said tables
-one error tooltip on hover
-one download all filtered button
-one download all pop-up (with 5 fields that are already created, and 2 new created (already counted))
-one setting in accounts settings allowing for quick download

12 changes or so, that would put your website at ultimate speed. 
Costing almost nothing. And saving a lot of people a lot of time. 

I can help you do it if you want.
Thanks for reopening the issue.
